packages/pigeon/test/kotlin_generator_test.dart

```diff
@@ -820,6 +942,24 @@ void main() {
             'val yArg = args[1].let { if (it is Int) it.toLong() else it as Long }'));
     expect(code, contains('wrapped["result"] = api.add(xArg, yArg)'));
     expect(code, contains('reply.reply(wrapped)'));
+    expect(code, contains('''
+        if (api != null) {
+          channel.setMessageHandler { message, reply ->
+            val wrapped = hashMapOf<String, Any?>()
+            try {
+              val args = message as List<Any?>
+              val xArg = args[0].let { if (it is Int) it.toLong() else it as Long }
+              val yArg = args[1].let { if (it is Int) it.toLong() else it as Long }
+              wrapped["result"] = api.add(xArg, yArg)
+            } catch (exception: Error) {
+              wrapped["error"] = wrapError(exception)
+            }
+            reply.reply(wrapped)
+          }
+        } else {
+          channel.setMessageHandler(null)
+        }
```

Comment:
There's a lot of redundant testing here; we should have specific, targeted tests for specific generation pieces, and only assert the relevant things in those tests. Testing a lot of unrelated things repeatedly makes test maintenance cumbersome and brittle.